neural_compressor/compression/pruner/__init__.py

```diff
@@ -19,7 +19,62 @@
 # model slim related
 from .model_slim.auto_slim import parse_auto_slim_config
 from .model_slim.auto_slim import model_slim
+from .pruning import PRUNINGS
+from .utils import process_config, torch, logger
+from typing import Optional, Union
 
+FRAMEWORK = {
+    'pytorch': 'pt',
+    'keras': 'keras'
+}
 
+def prepare_pruning(config, model: torch.nn.Module,
+                    opt_or_dataloader: Union[torch.optim.Optimizer, torch.utils.data.DataLoader],
```

Comment:
if the model is limited to torch.nn.module, why do we need a framework parameter? better detect the framework automatically